File: src/main/java/ir/syrent/velocityreport/spigot/event/PostReportEvent.java
package ir.syrent.velocityreport.spigot.event;

import ir.syrent.velocityreport.report.Report;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

public class PostReportEvent extends Event {

    private Report report;

    private static final HandlerList HANDLERS = new HandlerList();

    public PostReportEvent(Report report) {
        super(true);
        this.report = report;
    }

    public Report getReport() {
        return report;
    }

    @NotNull
    @Override
    public HandlerList getHandlers() {
        return HANDLERS;
    }

    public static HandlerList getHandlerList() {
        return HANDLERS;
    }
}

File: src/main/java/ir/syrent/velocityreport/spigot/event/PostReportEvent.kt
package ir.syrent.velocityreport.spigot.event

import ir.syrent.velocityreport.report.Report
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class PostReportEvent constructor(report: Report) : Event() {

    var report: Report

    init {
        this.report = report
    }

    override fun getHandlers(): HandlerList {
        return HANDLERS
    }

    companion object {
        private val HANDLERS = HandlerList()

        fun getHandlerList(): HandlerList {
            return HANDLERS
        }
    }

}